front-end/src/app/shared/resolvers/transaction.resolver.ts

```diff
@@ -70,14 +80,28 @@ export class TransactionResolver implements Resolve<Transaction | undefined> {
               );
             }
           } else {
+            // If there is a parent transaction
             if (transaction?.parent_transaction_id) {
               return this.transactionService.get(transaction.parent_transaction_id).pipe(
```

Comment:
I think we can safely use recursion here because we have a 3 tier cap.
```
return this.resolve_existing_transaction(transaction.parent_transaction_id).pipe(
     map((parent) => {
          transaction.parent_transaction = parent;
          return transaction;
```

Reply:
I'm not sure adding recursion is of all that much value when the depth is so shallow?  You could cut out a few lines of code, I guess.